examples/react-app/src/components/VideoPlayer.tsx

```diff
@@ -10,9 +11,22 @@ const videoData = dataSource.videoData({
 });
 
 const VideoPlayerComponent = () => {
+  let videoViewPort: Viewport = { duration: '0' };
+  // Get viewport from the time machine
+  const { viewport } = useViewport();
+  if (viewport) {
+    if (!isHistoricalViewport(viewport)) {
+      const start = new Date(Date.now() - parseDuration(viewport.duration));
+      const end = new Date();
+      videoViewPort = { start, end };
+    } else {
+      videoViewPort = viewport;
+    }
+  }
```

Comment:
`videoViewPort` should be `useMemo` that depend on `viewport`. the current logic will create a new instance of `{ duration: '0' }` during every rerender which will cause unnecessary rerender to VideoPlayer component

Reply:
Done - updated the code